flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/slowtaskdetector/ExecutionTimeBasedSlowTaskDetector.java

```diff
@@ -263,37 +257,38 @@ public void stop() {
         }
     }
 
-    /** This class defines the execution time and input number bytes for an execution. */
+    /** This class defines the execution time and input bytes for an execution. */
     @VisibleForTesting
-    static class ExecutionTimeWithInputNumBytes
-            implements Comparable<ExecutionTimeWithInputNumBytes> {
+    static class ExecutionTimeWithInputBytes implements Comparable<ExecutionTimeWithInputBytes> {
 
         private final long executionTime;
-        private final long inputNumBytes;
+        private final long inputBytes;
 
-        public ExecutionTimeWithInputNumBytes(long executionTime, long inputNumBytes) {
+        public ExecutionTimeWithInputBytes(long executionTime, long inputBytes) {
             this.executionTime = executionTime;
-            this.inputNumBytes = inputNumBytes;
+            this.inputBytes = inputBytes;
         }
 
         public long getExecutionTime() {
             return executionTime;
         }
 
-        public long getInputNumBytes() {
-            return inputNumBytes;
+        public long getInputBytes() {
+            return inputBytes;
         }
 
         @Override
-        public int compareTo(ExecutionTimeWithInputNumBytes other) {
-            // We should guarantee that all sorted elements' inputNumBytes are all zero or non-zero,
+        public int compareTo(ExecutionTimeWithInputBytes other) {
+            // We should guarantee that all sorted elements' inputBytes are all UNKNOWN or assigned,
             // otherwise it may cause ambiguity.
-            if (inputNumBytes == 0 || other.getInputNumBytes() == 0) {
+            if (inputBytes == NUM_BYTES_UNKNOWN || other.getInputBytes() == NUM_BYTES_UNKNOWN) {
```

Comment:
It's better to check that if one is UNKNOWN, the other is also UNKNOWN. Otherwise an exception should be thrown to expose this bug.